### PR TITLE
Capture CNI logs in istioctl bug-report

### PR DESCRIPTION
**Pull request by syw14**
Issue https://github.com/istio/istio/issues/45466
Fetch Cni logs from `kube-system` namespace pods selected by `k8s-app=istio-cni-node`. 

The logs are written into `cni.log` file per pod (not per container)

### CODE REVIEW

File: tools/bug-report/pkg/cluster/cluster.go [hunk 1]
```diff
@@ -230,6 +231,12 @@ func GetClusterResources(ctx context.Context, clientset *kubernetes.Clientset, c
 	}
 
 	for i, p := range pods.Items {
+		if p.Namespace == "kube-system" {
```
Comment (howardjohn) [inline tools/bug-report/pkg/cluster/cluster.go:234]: CNI can be in other namespaces (istio-system) is also common, maybe we do not check the namespace?

File: tools/bug-report/pkg/cluster/cluster.go [hunk 2]
```diff
@@ -298,8 +307,14 @@ func (r *Resources) insertContainer(namespace, deployment, pod, container string
 }
 
 // ContainerRestarts returns the number of container restarts for the given container.
-func (r *Resources) ContainerRestarts(namespace, pod, container string) int {
-	for _, cs := range r.Pod[PodKey(namespace, pod)].Status.ContainerStatuses {
+func (r *Resources) ContainerRestarts(namespace, pod, container string, isCniPod bool) int {
+	var podItem *corev1.Pod
+	if isCniPod {
```
Comment (howardjohn) [inline tools/bug-report/pkg/cluster/cluster.go:312]: do we need this? I thought r.Pod is a superset of r.CniPod

Comment (syw14) [reply tools/bug-report/pkg/cluster/cluster.go:312]: I added CniPod as a new field in `Resource` due to there's some pre defined [config](https://github.com/istio/istio/blob/1b2586b8a7739d47e7a0dc7e1e324e66e25e0a3a/tools/bug-report/pkg/filter/filter.go#L71) that will exclude all pods in namespace `kube-system` for `r.Pod` when building the "resource tree" of namespace/pod/container.

File: tools/bug-report/pkg/bugreport/bugreport.go [hunk 3]
```diff
@@ -443,7 +447,27 @@ func getOperatorLogs(runner *kubectlcmd.Runner, config *config.BugReportConfig,
...
+) {
+	wg.Add(1)
+	log.Infof("Waiting on CNI logs for %v", pod)
+	go func() {
+		defer func() {
+			wg.Done()
+			logRuntime(time.Now(), "Done getting CNI logs for %v", pod)
+		}()
+
+		clog, _, _, err := getLog(runner, resources, config, common.KubeSystemNamespace, pod, "")
```
Comment (hanxiaop) [inline tools/bug-report/pkg/bugreport/bugreport.go:467]: CNI may not be installed in `kube-system`

File: tools/bug-report/pkg/cluster/cluster.go (hunk 2 above)
Comment (howardjohn) [reply tools/bug-report/pkg/cluster/cluster.go:310]: I see... 

It seems like a bit weird to have CNI code in ContainerRestarts, a bit fragile. Maybe it makes sense to have an AllPods one that isn't filtered... 

Or we can keep this, not a big deal

**Review by howardjohn — APPROVED** [review]
LGTM! Supporting other namespace would be great, can add in a followup if you want

**Pull request merged**